File: Java/FactoryPattern_Tutorials/src/mypack/ShapeFactory.java
package mypack;

//Create a Factory to generate object of concrete class based on given information.

public class ShapeFactory {

	//use getShape method to get object of type shape
	public Shape getShape(String shapeType) {
		
		if(shapeType == null) {
			return null;
		}
		
		if(shapeType.equalsIgnoreCase("CIRCLE")) {
			return new Circle();
		}
		
		if(shapeType.equalsIgnoreCase("RECTANGLE")) {
			return new Rectangle();
		}
		
		if(shapeType.equalsIgnoreCase("SQUARE")) {
			return new Square();
		}
		
		return null;
	}
}

File: C#/TutorialsSpoint/AbstractFactory/ShapeFactory.cs
 class ShapeFactory : AbstractFactory
    {

        internal override Shape getShape(String shapeType)
        {
            if (shapeType.Equals("RECTANGLE", StringComparison.OrdinalIgnoreCase))
            {

                return new Rectangle();

            }
            if (shapeType.Equals("CIRCLE", StringComparison.OrdinalIgnoreCase))
            {

                return new Square();
            }
            return new Square();
        }
    }